src/codeu/chat/server/Transaction.java

```diff
@@ -0,0 +1,161 @@
+package codeu.chat.server;
+
+import codeu.chat.common.User;
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.util.Uuid;
+import codeu.chat.util.Tokenizer;
+import codeu.chat.util.Time;
+import codeu.chat.server.Controller;
+
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.FileWriter;
+import java.io.File;
+
+import java.io.IOException;
+
+
+public final class Transaction {
+
+  private ArrayList<String> transactionsList = new ArrayList<String>();
+  private Controller controller;
+
+  public Transaction(Controller controller) {
+    this.controller = controller;
+    read();
+  }
+
+  // takes in information that the server needs to populate itself!
+  private void read() {
+    File transaction = new File("transaction.txt");
+    try {
+      Scanner in = new Scanner(transaction);
+      while (in.hasNextLine()) {
+        String line = in.nextLine();
+        System.out.println(line);
+        Tokenizer tokens = new Tokenizer(line);
+        tokens.next();
+        switch (tokens.next()) {
+          case "CREATEUSER" : readUser(tokens); break;
+          case "CREATEMESSAGE" : readMessage(tokens); break;
+          case "CREATECONVO" : readConvo(tokens); break;
+        }
+      }
+    }
+    catch (IOException e) {
+    }
+  }
+
+  private void readUser(Tokenizer tokens) {
+    try {
+      tokens.next();
```

Comment:
While it's nice that your commands are human readable, the way they're currently implemented add a lot of these hard-to-undesrtand token.next() calls in your code.  Here's my recommendation.

1)  In writing the commands, every single variable should be encapsulated by starting and trailing ".  This should already be well handled by the Tokenizer.
2)  Right after the readLine() call grabbing the line from the file on disk, you should process the string before handing it off to the tokenizer.  For example, if the human-readiblity helper are all in the format of "<description>:", you should strip all of these strings out.  This way, the tokenizer will only return tokens that are relevant to your command.

In addition, since the commands are so specific for each type of action right now, I would add a comment for each read command that describes the expected format of the input.